apiml-security-common/src/main/java/org/zowe/apiml/security/common/error/ResourceAccessExceptionHandler.java

```diff
@@ -36,30 +35,30 @@ public ResourceAccessExceptionHandler(MessageService messageService, ObjectMappe
     /**
      * Entry method that takes care of an exception passed to it
      *
-     * @param request  Http request
-     * @param response Http response
-     * @param ex       Exception to be handled
-     * @throws ServletException Fallback exception if exception cannot be handled
+     * @param requestUri Http request URI
+     * @param addHeader Consumer to add response headers. This implementation does not use it
+     * @param ex         Exception to be handled
+     * @throws RuntimeException Fallback exception if exception cannot be handled
      */
     @Override
-    public void handleException(HttpServletRequest request, HttpServletResponse response, RuntimeException ex) throws ServletException {
+    public void handleException(String requestUri, BiConsumer<ApiMessageView, HttpStatus> function, BiConsumer<String, String> addHeader, RuntimeException ex) {
         if (ex instanceof GatewayNotAvailableException) {
-            handleGatewayNotAvailable(request, response, ex);
+            handleGatewayNotAvailable(requestUri, function, ex);
         } else if (ex instanceof ServiceNotAccessibleException) {
-            handleServiceNotAccessible(request, response, ex);
+            handleServiceNotAccessible(requestUri, function, ex);
         } else {
             throw ex;
         }
     }
 
-    //500
-    private void handleGatewayNotAvailable(HttpServletRequest request, HttpServletResponse response, RuntimeException ex) throws ServletException {
+    //503
+    private void handleGatewayNotAvailable(String requestUri, BiConsumer<ApiMessageView, HttpStatus> function, RuntimeException ex)  {
         log.debug(MESSAGE_FORMAT, HttpStatus.SERVICE_UNAVAILABLE.value(), ex.getMessage());
-        writeErrorResponse(ErrorType.GATEWAY_NOT_AVAILABLE.getErrorMessageKey(), HttpStatus.SERVICE_UNAVAILABLE, request, response);
+        writeErrorResponse(ErrorType.GATEWAY_NOT_AVAILABLE.getErrorMessageKey(), HttpStatus.SERVICE_UNAVAILABLE, requestUri, function);
     }
 
-    private void handleServiceNotAccessible(HttpServletRequest request, HttpServletResponse response, RuntimeException ex) throws ServletException {
+    private void handleServiceNotAccessible(String requestUri, BiConsumer<ApiMessageView, HttpStatus> function, RuntimeException ex) {
         log.debug(MESSAGE_FORMAT, HttpStatus.SERVICE_UNAVAILABLE.value(), ex.getMessage());
-        writeErrorResponse(ErrorType.SERVICE_UNAVAILABLE.getErrorMessageKey(), HttpStatus.SERVICE_UNAVAILABLE, request, response);
+        writeErrorResponse(ErrorType.SERVICE_UNAVAILABLE.getErrorMessageKey(), HttpStatus.SERVICE_UNAVAILABLE, requestUri, function);
```

Comment:
```suggestion
    public void handleException(String requestUri, BiConsumer<ApiMessageView, HttpStatus> function, BiConsumer<String, String> addHeader, RuntimeException ex) {
        ErrorType errorType;
        if (ex instanceof GatewayNotAvailableException) {
            errorType = ErrorType.GATEWAY_NOT_AVAILABLE;
        } else if (ex instanceof ServiceNotAccessibleException) {
            errorType = ErrorType.SERVICE_UNAVAILABLE;
        } else {
            throw ex;
        }
        
        log.debug(MESSAGE_FORMAT, HttpStatus.SERVICE_UNAVAILABLE.value(), ex.getMessage());
        writeErrorResponse(errorType.getErrorMessageKey(), HttpStatus.SERVICE_UNAVAILABLE, requestUri, function);
    }
```

Just if you consider it readable. It should be definitely more maintanable.

Reply:
I don't see much difference here, but ok...